app/controllers/user_groups_controller.rb

```diff
@@ -66,12 +69,28 @@ def remove_users
   end
 
   def possible_groups
-    @user_groups = @user_groups.name_matching(params[:q])
-    render json: @user_groups
+    if params[:select2].present?
+      possible_groups_select2
+    else
+      @user_groups = @user_groups.name_matching(params[:q])
+      render(json: @user_groups)
+    end
   end
 
   private
 
+  def possible_groups_select2
+    @user_groups = @user_groups.name_matching(params[:search]) if params[:search].present?
+    @user_groups = @user_groups
+      .natural_sort_by_key
+      .paginate(page: params[:page], per_page: 20)
```

Comment:
Style-wise here I think it's better to do like:

```
x = y.foo
  .bar
```

because it's one less line, meaning you're less likely to run afoul of the method line limit.